javav2/usecases/pam_source_files/src/main/java/com/example/photo/endpoints/RestoreEndpoint.java

```diff
@@ -24,21 +24,31 @@ public RestoreEndpoint(DynamoDBService dynamoDBService, S3Service s3Service, Sns
     }
 
     public Job restore(String notify, List<String> tags) {
-        String topicArn = this.snsService.createNotificationTopic(notify);
-
         Set<String> images = tags.stream().parallel().flatMap(this::imagesByTag).collect(Collectors.toSet());
         String manifest = makeManifest(PhotoApplicationResources.STORAGE_BUCKET, images);
         String manifestArn = this.s3Service.putManifest(manifest);
-        String jobId = this.s3Service.startRestore(manifestArn, tags);
 
-        Job job = new Job(jobId, topicArn);
+        // These three values are required for createJob() to work: etag. accountID, roleARN!
+        String eTag = this.s3Service.getETag(manifestArn, PhotoApplicationResources.WORKING_BUCKET);
+
+        // This needs to be pulled in from CDK!
+        String accountID = "924034042979";  // REPLACE WITH ACCOUNT ID
+
+        // CDK should create this too.
+        String roleARN = "arn:aws:iam::924034042979:role/pamRole";  // Replace with valid IAM ROLE
```

Comment:
Let's add this role to the CDK, and then export its ARN to the lambda's environment.

Reply:
Not needed